server/channels/web/webhook.go

```diff
@@ -86,14 +86,17 @@ func incomingWebhook(c *Context, w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	appErr = c.App.HandleIncomingWebhook(c.AppContext, id, incomingWebhookPayload)
+	post, appErr := c.App.HandleIncomingWebhook(c.AppContext, id, incomingWebhookPayload)
 	if appErr != nil {
 		c.Err = model.NewAppError("incomingWebhook", "web.incoming_webhook.general.app_error", errCtx, "", appErr.StatusCode).Wrap(appErr)
 		return
 	}
 
-	w.Header().Set("Content-Type", "text/plain")
-	w.Write([]byte("ok"))
+	w.Header().Set("Content-Type", "application/json")
+
+	if err := post.EncodeJSON(w); err != nil {
+		c.Logger.Warn("Error while writing response", mlog.Err(err))
```

Comment:
The general pattern here seems to be to set `c.Err` on error and return - then handle the error in the defer function:
https://github.com/mattermost/mattermost/blob/3a337479dcc9a2426874e552ea4906d661c76da1/server/channels/web/webhook.go#L46-L48
Should we follow that same pattern in this case?

Reply:
I think what you have is fine, especially if this is what we do elsewhere. Thanks :+1: